src/main/java/org/folio/uk/service/UserService.java

```diff
@@ -98,21 +108,50 @@ public Optional<User> getUser(UUID id) {
     return usersClient.lookupUserById(id);
   }
 
-  public CompositeUser getUserBySelfReference(List<IncludedField> include, boolean expandPermissions) {
+  public CompositeUser getUserBySelfReference(List<IncludedField> include, boolean expandPermissions,
+                                              boolean overrideUser) {
     log.info("Retrieving user by self reference with parameters: include = {}, expandPermissions = {}",
       () -> StringUtils.join(emptyIfNull(include), ", "), () -> expandPermissions);
 
     var userId = getUserId();
-
     var user = usersClient.lookupUserById(userId)
       .orElseThrow(() -> new EntityNotFoundException("User was Not Found with: id = " + userId));
 
-    return new CompositeUser()
-      .user(user)
+    // TODO Switch overrideUser back to assert true
+    if (Boolean.TRUE.equals(keycloakFederatedAuthProperties.isEnabled()) && Boolean.FALSE.equals(overrideUser)) {
```

Comment:
can be simplified, UI during login will send always overrideUser=true regardless of any env variable or feature flags.
In BE we can check if overrideUser=true and userType=shadow in this case try to ovveride. 
So in this case for non-ECS login we will never have shadow user.
For cases when Keycloak is not properly configured - we also will never have shadow user type, because could not login.

Just lets write small comment why we need this ovveride logic and add more description to this request param.